flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -937,6 +938,15 @@ public CompletableFuture<ExecutionGraphInfo> requestExecutionGraphInfo(
                 .exceptionally(checkExecutionGraphStoreOnException);
     }
 
+    @Override
+    public CompletableFuture<CheckpointStatsSnapshot> requestCheckpointStats(
+            JobID jobId, Time timeout) {
+        Optional<JobManagerRunner> maybeJob = getJobManagerRunner(jobId);
+
+        return maybeJob.map(job -> job.requestCheckpointStats(timeout))
+                .orElse(FutureUtils.completedExceptionally(new FlinkJobNotFoundException(jobId)));
```

Comment:
`org.apache.flink.runtime.dispatcher.Dispatcher#performOperationOnJobMasterGateway` should be used here instead